File: Top150Interview/C#/121_Best_Time_To_Buy_Sell_Stocks.cs
public class Solution
{
    public int MaxProfit(int[] prices)
    {
        int left = 0;   // buy day
        int right = 1;  // sell day
        int maxP = 0;

        while (right < prices.Length)
        {
            if (prices[left] < prices[right])
            {
                int profit = prices[right] - prices[left];
                if (profit > maxP) maxP = profit;
            }
            else
            {
                left = right;
            }
            right++;
        }
        return maxP;
    }
}

File: Top150Interview/Python/121_Best_Time_To_Buy_Sell_Stocks.py
class Solution(object):
    def maxProfit(self, prices):
        left, right = 0, 1
        maxP = 0
        
        while right < len(prices):
            if prices[left] < prices[right]:
                profit = prices[right] - prices[left]
                maxP = max(maxP, profit)
            else:
                left = right
            right += 1

        return maxP